cisagov/manage.get.gov
# Issue #2541 additional domain request dates [getgov-litterbox]

## Ticket

Resolves #2541 


## Changes


- Added two new fields to the domain request model, and renamed submission_date to last_submitted_date
- Added appropriate migration script
- Added new script to populate the new fields for old data
- Added logic to update the new date fields as appropriate
- refactored references to the old submission_date field
- Updated tests to work with the new fields



## Context for reviewers



## Setup


The code is currently on getgov-litterbox

## Code Review Verification Steps
This is a bit tricky to review because the way our test data is generated on sandboxes doesn't leave the domains with any history to look through. Since auditlogs are used by the populate script, the only real way to do this is by hand.
1. Open [getgov-litterbox](getgov-litterbox.app.cloud.gov)
2. navigate to the admin page
3. open the domain request model and observe the new fields, they will be empty initially
4. Take any existing domain request and submit it, and then note how the fields update
5. Run the populate script using the command `python manage.py populate_domain_request_dates`
6. Observe how (some) of the dates for the old data have now been filled in. Some dates will still be missing because our test data does not have history from which to grab previous dates. To double check this, open some domain request models and check their history. If there is a submitted action previously, then check the date against what was populated in the new fields.

### As the original developer, I have

#### Satisfied acceptance criteria and met development standards


#### Ensured code standards are met (Original Developer)


#### Validated user-facing changes (if applicable)


### As a code reviewer, I have

#### Reviewed, tested, and left feedback about the changes


#### Ensured code standards are met (Code reviewer)


#### Validated user-facing changes as a developer




**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

### As a designer reviewer, I have

#### Verified that the changes match the design intention


#### Validated user-facing changes as a designer




## Screenshots
